examples/vector_databases/tair/QA_with_Langchain_Tair_and_OpenAI.ipynb

```diff
@@ -0,0 +1,367 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Question Answering with Langchain, Tair and OpenAI\n",
+    "This notebook presents how to implement a Question Answering system with Langchain, Tair as a knowledge based and OpenAI embeddings. If you are not familiar with Tair, it’s better to check out the [Getting_started_with_Tair_and_OpenAI.ipynb](Getting_started_with_Tair_and_OpenAI.ipynb) notebook.\n",
+    "\n",
+    "This notebook presents an end-to-end process of:\n",
+    "- Calculating the embeddings with OpenAI API.\n",
+    "- Storing the embeddings in an Tair instance to build a knowledge base.\n",
+    "- Converting raw text query to an embedding with OpenAI API.\n",
+    "- Using Tair to perform the nearest neighbour search in the created collection to find some context.\n",
+    "- Asking LLM to find the answer in a given context.\n",
+    "\n",
+    "All the steps will be simplified to calling some corresponding Langchain methods."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "For the purposes of this exercise we need to prepare a couple of things:\n",
+    "[Tair cloud instance](https://www.alibabacloud.com/help/en/tair/latest/what-is-tair).\n",
+    "[Langchain](https://github.com/hwchase17/langchain) as a framework.\n",
+    "An OpenAI API key."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install requirements\n",
+    "This notebook requires the following Python packages: `openai`, `tiktoken`, `langchain` and `tair`.\n",
+    "- `openai` provides convenient access to the OpenAI API.\n",
+    "- `tiktoken` is a fast BPE tokeniser for use with OpenAI's models.\n",
+    "- `langchain` helps us to build applications with LLM more easily.\n",
+    "- `tair` library is used to interact with the tair vector database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2023-05-06T10:21:40.843630Z",
+     "start_time": "2023-05-06T10:21:38.796769Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "! pip install openai tiktoken langchain tair "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare your OpenAI API key\n",
+    "The OpenAI API key is used for vectorization of the documents and queries.\n",
+    "\n",
+    "If you don't have an OpenAI API key, you can get one from [https://platform.openai.com/account/api-keys ).\n",
+    "\n",
+    "Once you get your key, please add it to your environment variables as `OPENAI_API_KEY` by running following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2023-05-06T10:21:40.974668Z",
+     "start_time": "2023-05-06T10:21:40.845980Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "! export OPENAI_API_KEY=\"your API key\""
```

Comment:
We generally don't want to encourage people writing their OpenAI key in a code block. Instead, use something like `getpass`

Reply:
Thanks for your suggestions! We will change the OpenAI key input way to getpass.